lib/AST/Type.cpp

```diff
@@ -2027,9 +2027,14 @@ static ForeignRepresentableKind
 getObjCObjectRepresentable(Type type, const DeclContext *dc) {
   // @objc metatypes are representable when their instance type is.
   if (auto metatype = type->getAs<AnyMetatypeType>()) {
+    auto instanceType = metatype->getInstanceType();
+
+    // Consider protocol metatypes as not Objective-C representable.
+    if (metatype->is<MetatypeType>() && instanceType->is<ProtocolType>())
+      return ForeignRepresentableKind::None;
```

Comment:
Y eso?

Enviado desde mi iPhone

> El ago 5, 2016, a las 1:11 PM, John McCall <notifications@github.com> escribió:
> 
> In lib/AST/Type.cpp:
> 
> > @@ -2027,9 +2027,14 @@ static ForeignRepresentableKind
> >  getObjCObjectRepresentable(Type type, const DeclContext *dc) {
> >    // @objc metatypes are representable when their instance type is.
> >    if (auto metatype = type->getAs<AnyMetatypeType>()) {
> > +    auto instanceType = metatype->getInstanceType();
> > +
> > +    // Consider protocol metatypes as not Objective-C representable.
> > +    if (metatype->is<MetatypeType>() && instanceType->is<ProtocolType>())
> > +      return ForeignRepresentableKind::None;
> Well, protocol compositions (id) and existential metatypes (Class
> 
> ) should be ObjC-representable if their protocols are all @objc, so if that's the condition, they'd get by.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>